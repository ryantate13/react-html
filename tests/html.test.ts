import {html} from '../src';

describe('html', () => {
    it('renders with children', () => {
        expect(html({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(html({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(html({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(html()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(html({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(html({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(html({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(html({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(html({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(html({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(html({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(html({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(html({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(html({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(html({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(html({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(html({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(html({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(html({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(html({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(html({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(html({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(html({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(html({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(html({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(html({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(html({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(html({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(html({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(html({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(html({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(html({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(html({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(html({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(html({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(html({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(html({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(html({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(html({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(html({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(html({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(html({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(html({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(html({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(html({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(html({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(html({translate: "no"})).toMatchSnapshot();
    });

    it('renders with manifest="https://google.com/"', () => {
        expect(html({manifest: "https://google.com/"})).toMatchSnapshot();
    });
});