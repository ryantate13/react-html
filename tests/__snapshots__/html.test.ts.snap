// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`html renders arrays of children 1`] = `
<html>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</html>
`;

exports[`html renders with accessKey="a,b,c,d" 1`] = `
<html
  accessKey="a,b,c,d"
/>
`;

exports[`html renders with autoCapitalize="characters" 1`] = `
<html
  autoCapitalize="characters"
/>
`;

exports[`html renders with autoCapitalize="none" 1`] = `
<html
  autoCapitalize="none"
/>
`;

exports[`html renders with autoCapitalize="off" 1`] = `
<html
  autoCapitalize="off"
/>
`;

exports[`html renders with autoCapitalize="on" 1`] = `
<html
  autoCapitalize="on"
/>
`;

exports[`html renders with autoCapitalize="sentences" 1`] = `
<html
  autoCapitalize="sentences"
/>
`;

exports[`html renders with autoCapitalize="words" 1`] = `
<html
  autoCapitalize="words"
/>
`;

exports[`html renders with children 1`] = `
<html>
  TEST
</html>
`;

exports[`html renders with className="foo bar baz" 1`] = `
<html
  className="foo bar baz"
/>
`;

exports[`html renders with contentEditable="false" 1`] = `
<html
  contentEditable="false"
/>
`;

exports[`html renders with contentEditable="true" 1`] = `
<html
  contentEditable="true"
/>
`;

exports[`html renders with dir="auto" 1`] = `
<html
  dir="auto"
/>
`;

exports[`html renders with dir="ltr" 1`] = `
<html
  dir="ltr"
/>
`;

exports[`html renders with dir="rtl" 1`] = `
<html
  dir="rtl"
/>
`;

exports[`html renders with draggable="false" 1`] = `
<html
  draggable="false"
/>
`;

exports[`html renders with draggable="true" 1`] = `
<html
  draggable="true"
/>
`;

exports[`html renders with empty attributes 1`] = `<html />`;

exports[`html renders with hidden=false 1`] = `
<html
  hidden={false}
/>
`;

exports[`html renders with hidden=true 1`] = `
<html
  hidden={true}
/>
`;

exports[`html renders with id="TEST" 1`] = `
<html
  id="TEST"
/>
`;

exports[`html renders with inputMode="decimal" 1`] = `
<html
  inputMode="decimal"
/>
`;

exports[`html renders with inputMode="email" 1`] = `
<html
  inputMode="email"
/>
`;

exports[`html renders with inputMode="none" 1`] = `
<html
  inputMode="none"
/>
`;

exports[`html renders with inputMode="numeric" 1`] = `
<html
  inputMode="numeric"
/>
`;

exports[`html renders with inputMode="search" 1`] = `
<html
  inputMode="search"
/>
`;

exports[`html renders with inputMode="tel" 1`] = `
<html
  inputMode="tel"
/>
`;

exports[`html renders with inputMode="text" 1`] = `
<html
  inputMode="text"
/>
`;

exports[`html renders with inputMode="url" 1`] = `
<html
  inputMode="url"
/>
`;

exports[`html renders with is="foo" 1`] = `
<html
  is="foo"
/>
`;

exports[`html renders with itemID="https://google.com/" 1`] = `
<html
  itemID="https://google.com/"
/>
`;

exports[`html renders with itemProp="https://google.com/" 1`] = `
<html
  itemProp="https://google.com/"
/>
`;

exports[`html renders with itemRef="foo bar" 1`] = `
<html
  itemRef="foo bar"
/>
`;

exports[`html renders with itemScope=false 1`] = `
<html
  itemScope={false}
/>
`;

exports[`html renders with itemScope=true 1`] = `
<html
  itemScope={true}
/>
`;

exports[`html renders with itemType="https://google.com/" 1`] = `
<html
  itemType="https://google.com/"
/>
`;

exports[`html renders with lang="" 1`] = `
<html
  lang=""
/>
`;

exports[`html renders with lang="es-419" 1`] = `
<html
  lang="es-419"
/>
`;

exports[`html renders with slot="TEST" 1`] = `
<html
  slot="TEST"
/>
`;

exports[`html renders with spellCheck="false" 1`] = `
<html
  spellCheck="false"
/>
`;

exports[`html renders with spellCheck="true" 1`] = `
<html
  spellCheck="true"
/>
`;

exports[`html renders with style={"color":"#000000"} 1`] = `
<html
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`html renders with tabIndex=1 1`] = `
<html
  tabIndex={1}
/>
`;

exports[`html renders with title="TEST" 1`] = `
<html
  title="TEST"
/>
`;

exports[`html renders with translate="no" 1`] = `
<html
  translate="no"
/>
`;

exports[`html renders with translate="yes" 1`] = `
<html
  translate="yes"
/>
`;

exports[`html renders without attributes 1`] = `<html />`;
